src/charm.py

```diff
@@ -422,6 +420,11 @@ def _common_exit_hook(self, update_ca_certs: bool = False) -> None:
 
         self.alertmanager_provider.update_relation_data()
 
+        self.ingress.provide_ingress_requirements(
+            scheme=urlparse(self._internal_url).scheme, port=self.api_port
```

Comment:
worth wrapping `urlparse(self._internal_url).scheme` in a property?

Reply:
It only appears twice so far. `urlparse().scheme` seems straightforward enough to keep "inlined" at this point I think.